pkg/actors/reminders/scheduler.go

```diff
@@ -24,48 +24,73 @@ import (
 	"google.golang.org/protobuf/types/known/wrapperspb"
 
 	"github.com/dapr/dapr/pkg/actors/internal"
+	"github.com/dapr/dapr/pkg/actors/reminders/migration"
 	apierrors "github.com/dapr/dapr/pkg/api/errors"
+	"github.com/dapr/dapr/pkg/healthz"
 	schedulerv1pb "github.com/dapr/dapr/pkg/proto/scheduler/v1"
 	"github.com/dapr/dapr/pkg/runtime/scheduler/clients"
 	"github.com/dapr/kit/ptr"
 	kittime "github.com/dapr/kit/time"
 )
 
 type SchedulerOptions struct {
-	Namespace string
-	AppID     string
-	Clients   *clients.Clients
+	Namespace        string
+	AppID            string
+	Clients          *clients.Clients
+	ProviderOpts     internal.ActorsProviderOptions
+	ListActorTypesFn func() []string
+	Healthz          healthz.Healthz
 }
 
 // Implements a reminders provider that does nothing when using Scheduler Service.
 type scheduler struct {
-	namespace string
-	appID     string
-	clients   *clients.Clients
+	namespace        string
+	appID            string
+	clients          *clients.Clients
+	lookUpActorFn    internal.LookupActorFn
+	stateReminder    internal.RemindersProvider
+	listActorTypesFn func() []string
+	htarget          healthz.Target
 }
 
 func NewScheduler(opts SchedulerOptions) internal.RemindersProvider {
 	return &scheduler{
-		clients:   opts.Clients,
-		namespace: opts.Namespace,
-		appID:     opts.AppID,
+		clients:          opts.Clients,
+		namespace:        opts.Namespace,
+		appID:            opts.AppID,
+		stateReminder:    NewStateStore(opts.ProviderOpts),
+		listActorTypesFn: opts.ListActorTypesFn,
+		htarget:          opts.Healthz.AddTarget(),
 	}
 }
 
 func (s *scheduler) SetExecuteReminderFn(fn internal.ExecuteReminderFn) {
 }
 
 func (s *scheduler) SetStateStoreProviderFn(fn internal.StateStoreProviderFn) {
+	s.stateReminder.SetStateStoreProviderFn(fn)
 }
 
 func (s *scheduler) SetLookupActorFn(fn internal.LookupActorFn) {
+	s.lookUpActorFn = fn
+	s.stateReminder.SetLookupActorFn(fn)
 }
 
 func (s *scheduler) SetMetricsCollectorFn(fn remindersMetricsCollectorFn) {
```

Comment:
should we fill in this func too while you are in this file? also fine to do as a follow up